### ISSUE TITLE
Update MeanComparison so it always uses float64 precision for its sum

### ISSUE THREAD

**Issue by gideonite** [ai2cm/ace#380]
> Not part of this PR, but you should consider updating MeanComparison so it always uses float64 precision for its sum by setting the dtype in `self._sum = {k: torch.zeros_like(v) for k, v in metric.items()}`. There's a risk of fairly significant roundoff error if the aggregator is used over many batches at float32 precision. You can make an issue referencing/copying this comment.

_Originally posted by @mcgibbon in https://github.com/ai2cm/full-model/pull/969#discussion_r1666986344_
